app/controllers/coronavirus_form/product_details_controller.rb

```diff
@@ -78,27 +82,67 @@ def validate_missing_fields
     required.each_with_object([]) do |field, invalid_fields|
       next if @product[field.to_sym].present?
 
-      invalid_fields << {
-        field: field,
-        text: t("coronavirus_form.questions.#{controller_name}.#{field}.custom_error",
-                default: t("coronavirus_form.errors.missing_mandatory_text_field",
-                           field: t("coronavirus_form.questions.#{controller_name}.#{field}.label")).humanize),
-      }
+      invalid_fields << error_response(field)
     end
   end
 
+  def validate_product_quantity
+    field = "product_quantity"
+    return error_response(field) unless @product[field.to_sym]
+
+    quantity = Integer(@product[field.to_sym])
+    return if quantity && quantity.positive?
+
+    error_response(field)
+  rescue ArgumentError
```

Comment:
Looks better, thanks 👍

Perhaps we could do a `.gsub(/[a-zA-Z]| /, "")` to try and prevent argument errors, but it might bring up the same issues as mentioned before. Attempting to correct the user might confuse things when they get asked about their "10100" items but they wrote "between 10 and 100".

We could also try using `.to_i` here, which gives `0` as a fallback, rather than raising an error. Would want to wrap all this logic in a kind of `UserProvidedInteger` class, but creating all this feels a bit like building GOV.UK Submit :)

I thought way too long about it - this seems good to me 👍